modules/core/src/main/java/org/apache/ignite/internal/util/HashCalculator.java

```diff
@@ -170,8 +244,19 @@ public void appendNumber(BigInteger v) {
      * @param v Value to update hash.
      */
     public void appendUuid(UUID v) {
-        appendLong(v.getMostSignificantBits());
-        appendLong(v.getLeastSignificantBits());
+        hash = combine(hash, hashUuid(v));
+    }
+
+    /**
+     * Get value hash.
+     *
+     * @param v Value to hash.
+     */
+    public static int hashUuid(UUID v) {
+        int hash1 = hashLong(v.getMostSignificantBits());
+        int hash2 = hashLong(v.getLeastSignificantBits());
+
+        return combine(hash1, hash2);
```

Comment:
Do we need to use `combine` here? I'd avoid using it where it is possible.
```suggestion
        int hash = HashUtils.hash32(v.getMostSignificantBits());
        hash = HashUtils.hash32(v.getLeastSignificantBits(), hash);

        return hash;
```

Reply:
Good point, fixed.